showcase/app/controllers/page-components/copy/button.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { scheduleOnce } from '@ember/runloop';
+
+import type Owner from '@ember/owner';
+import type RouterService from '@ember/routing/router-service';
+
+import type { PageComponentsCopyButtonModel } from 'showcase/routes/page-components/copy/button';
+
+export default class PageComponentsCopyButtonController extends Controller {
+  declare model: PageComponentsCopyButtonModel;
+  @service declare router: RouterService;
+  @tracked isModalActive = false;
+
+  constructor(owner: Owner) {
+    super(owner);
+    this.router.on('routeDidChange', this.routeDidChange.bind(this));
+  }
+
+  routeDidChange() {
+    if (this.router.currentRoute?.name === 'page-components.copy.button') {
+      // eslint-disable-next-line ember/no-runloop
+      scheduleOnce('afterRender', this, this.replaceMockCopyStatus.bind(this));
+    }
+  }
+
+  replaceMockCopyStatus() {
```

Comment:
[question] in the `showcase/app/controllers/page-components/pagination/index.ts` the functions `getCursorParts` and `getNewPrevNextCursors` have been left outside of the main class, while here `replaceMockCopyStatus` has been moved inside. What is the reason/difference between the two approaches? in which case one should be used vs the other?

Reply:
I moved it into the main class in this case because it was using `this`, the pagination functions don't need to access  it. For typing it was just a lot simpler to move the function into the main class.